app/services/topics/mutator.rb

```diff
@@ -55,6 +55,7 @@ def mutate
       attach_files(document_signed_ids)
       # topic.persisted? means that we are updating existing topic and documents can be removed
       shadow_delete_documents(docs_to_delete) if topic.persisted?
+      rename_files(document_signed_ids)
```

Comment:
Will this leave S3 files with original names?
If yes, will this mismatch cause any troubles for us?